# Machine Translation





## Links to Models
##### Google Translate: https://cloud.google.com/translate?hl=en & https://github.com/ahmeterenodaci/easygoogletranslate
##### Azure AI Translator: https://azure.microsoft.com/en-us/products/ai-services/ai-translator
##### MyMemory: https://mymemory.translated.net/doc/spec.php


In [124]:
import time

In [125]:
from google.colab import drive
drive.mount('/content/Drive')

Drive already mounted at /content/Drive; to attempt to forcibly remount, call drive.mount("/content/Drive", force_remount=True).


In this code, I have translated Text from English to Spanish But it has the capability to translate in almost all languages.

##Translating Text to Spanish using Google Translate

In [126]:
# Installing the 'easygoogletranslate' library
!pip install easygoogletranslate

In [127]:
# Importing required modules
from easygoogletranslate import EasyGoogleTranslate

start_time = time.time()

translator = EasyGoogleTranslate(
    source_language='en',
    target_language='es',
    timeout=10
)
# Enter the text to be translated
result = translator.translate(" Hello there! My name is Phil. I have been an online English tutor for almost three years now. I've been helping students from beginner to advanced level improve their English skills. I've tried teaching kids and adults. I used CPR. Look! Rewards! And many more! I would love to help you improve and make progress in learning. From phonics, grammar, pronunciation and up to conversational English. I've already earned my Tassel and Tia Felt certificate, so you're in good hands. Learn with me and let's make English a fun and easy language. See ya!")

print(result)
print("Google Translate : %s seconds" % (time.time() - start_time))

¡Hola! Mi nombre es Fil. He sido tutor de inglés en línea durante casi tres años. He estado ayudando a estudiantes desde nivel principiante hasta avanzado a mejorar sus habilidades en inglés. He intentado enseñar a niños y adultos. Utilicé RCP. ¡Mirar! ¡Recompensas! ¡Y muchos más! Me encantaría ayudarte a mejorar y progresar en el aprendizaje. Desde fonética, gramática, pronunciación y hasta inglés conversacional. Ya obtuve mi certificado de Borla y Tia Felt, así que estás en buenas manos. Aprende conmigo y hagamos del inglés un idioma fácil y divertido. ¡Nos vemos!
Google Translate : 0.3525381088256836 seconds


##Translating Text to Spanish using Azure AI Translator

In [128]:
# Install the 'requests' and 'uuid' libraries
!pip install requests uuid

In [129]:
# Import required modules
import requests

start_time = time.time()

# Define your Azure Cognitive Services Translator API credentials and configuration
key = "ebe0efc466cd4c119e987d8fb0a7c748"
endpoint = "https://api.cognitive.microsofttranslator.com/"

# Define the Azure region
location = "westus2"

# Define the API path for translation
path = '/translate'
constructed_url = endpoint + path

# Define parameters
params = {
    'api-version': '3.0',
    'from': 'en',
    'to': ['es']
}

# Define request headers
headers = {
    'Ocp-Apim-Subscription-Key': key,
    'Ocp-Apim-Subscription-Region': location,
    'Content-type': 'application/json',
    'X-ClientTraceId': str(uuid.uuid4())
}

# Define the text to be translated
body = [{
    'text': "Hello there! My name is Phil. I have been an online English tutor for almost three years now. I've been helping students from beginner to advanced level improve their English skills. I've tried teaching kids and adults. I used CPR. Look! Rewards! And many more! I would love to help you improve and make progress in learning. From phonics, grammar, pronunciation and up to conversational English. I've already earned my Tassel and Tia Felt certificate, so you're in good hands. Learn with me and let's make English a fun and easy language. See ya!"
}]

# Make a POST request
request = requests.post(constructed_url, params=params, headers=headers, json=body)

response = request.json()

translation = response[0]['translations'][0]['text']

print(translation)

print("Azure AI Translator: %s seconds" % (time.time() - start_time))


¡Hola! Mi nombre es Phil. He sido tutor de inglés en línea durante casi tres años. He estado ayudando a estudiantes desde principiantes hasta niveles avanzados a mejorar sus habilidades en inglés. He intentado enseñar a niños y adultos. Usé RCP. ¡Mirar! ¡Recompensas! ¡Y muchos más! Me encantaría ayudarte a mejorar y progresar en el aprendizaje. Desde la fonética, la gramática, la pronunciación y hasta el inglés conversacional. Ya obtuve mi certificado de Tassel y Tia Felt, así que estás en buenas manos. Aprende conmigo y hagamos del inglés un idioma fácil y divertido. ¡Nos vemos!
Azure AI Translator: 0.4350011348724365 seconds


##Translating Text to Spanish using MyMemory

In [130]:
# Install the 'mtranslate' library
!pip install mtranslate

In [131]:
# Import the 'mtranslate' library
from mtranslate import translate

start_time = time.time()

def translate_to_spanish(text):
    try:
        translated_text = translate(text, 'es', 'en')
        return translated_text
    except Exception as e:
        return f"Translation error: {str(e)}"

if __name__ == "__main__":
    # Define the text to be translated
    text_to_translate = " Hello there! My name is Phil. I have been an online English tutor for almost three years now. I've been helping students from beginner to advanced level improve their English skills. I've tried teaching kids and adults. I used CPR. Look! Rewards! And many more! I would love to help you improve and make progress in learning. From phonics, grammar, pronunciation and up to conversational English. I've already earned my Tassel and Tia Felt certificate, so you're in good hands. Learn with me and let's make English a fun and easy language. See ya!"
    translated_text = translate_to_spanish(text_to_translate)
    print(translated_text)


print("MyMemory : %s seconds" % (time.time() - start_time))

¡Hola! Mi nombre es Fil. He sido tutor de inglés en línea durante casi tres años. He estado ayudando a estudiantes desde nivel principiante hasta avanzado a mejorar sus habilidades en inglés. He intentado enseñar a niños y adultos. Utilicé RCP. ¡Mirar! ¡Recompensas! ¡Y muchos más! Me encantaría ayudarte a mejorar y progresar en el aprendizaje. Desde fonética, gramática, pronunciación y hasta inglés conversacional. Ya obtuve mi certificado de Borla y Tia Felt, así que estás en buenas manos. Aprende conmigo y hagamos del inglés un idioma fácil y divertido. ¡Nos vemos!
MyMemory : 0.4078376293182373 seconds


### Time Taken to Translate by Different models

| Model    | Time of Execution (in sec)|
| :---        |    :----:   
| Google Translate   | 0.35    |
| Azure AI Translator | 0.43     |
| MyMemory  | 0.40        |